### ISSUE TITLE
ty: dict should be compatible with Mapping

### ISSUE THREAD

**Issue by JelleZijlstra** [astral-sh/ruff#17904]
### Summary

ty fails to recognize that a `dict` (i.e., `dict[Any, Any]`) is assignable to `Mapping[str, int]`.

```python
from collections.abc import Mapping

def capybara(x: Mapping[str, int]): ...

def hutia(x: dict):
    capybara(x)
```

```
    Argument to this function is incorrect: Expected `Mapping[str, int]`, found `dict[Unknown, Unknown]` (lint:invalid-argument-type) [Ln 6, Col 14]
```

https://play.ty.dev/b945b5e0-9b04-4e28-93c2-76ba33b24f9f

Not sure what exactly is going on here; I thought it might be something to do with variance but it does accept `list` for `MutableSequence[int]`.

**Comment by carljm** [comment]
It works with `dict[str, int]` but fails with `dict[Any, Any]`.

cc @dcreager

**Comment by JelleZijlstra** [comment]
Noticed another one that looks similar:

```
error: lint:invalid-assignment: Object of type `defaultdict[Unknown, Unknown]` is not assignable to `dict[type[ADT], _TagCount]`
    --> taxonomy/shell.py:3049:5
     |
3047 |         return
3048 |
3049 |     counts: dict[type[ADT], _TagCount] = defaultdict(_TagCount)
     |     ^^^^^^
3050 |     for obj in model.select_valid():
3051 |         seen_for_nam = set()
```

Also seeing this with `Counter`.

**Comment by charliermarsh** [comment]
Is this not the same as https://github.com/astral-sh/ruff/issues/17817?

**Comment by JelleZijlstra** [comment]
Yes it is
